lib/ansible/modules/cloud/amazon/ec2_customer_gateway_facts.py

```diff
@@ -0,0 +1,155 @@
+                "customer_gateway_name": "test-customer-gw",
+                "ip_address": "110.112.113.120",
+                "state": "available",
+                "tags": [
+                    {
+                        "key": "Name",
+                        "value": "test-customer-gw"
+                    }
+                ],
+                "type": "ipsec.1"
+            }
+        ]
+changed:
+    description: True, if listing of customer gateways succeeds.
+    returned: always
+    type: bool
+    sample: "false"
+'''
+
+import traceback
+import json
+try:
+    import boto3
+    from botocore.exceptions import ClientError
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import (ansible_dict_to_boto3_filter_list,
+                                      boto3_conn, boto3_tag_list_to_ansible_dict, camel_dict_to_snake_dict,
+                                      ec2_argument_spec, get_aws_connection_info)
+
+
+def date_handler(obj):
+    return obj.isoformat() if hasattr(obj, 'isoformat') else obj
+
+
+def list_customer_gateways(connection, module):
+    params = dict()
+
+    params['Filters'] = ansible_dict_to_boto3_filter_list(module.params.get('filters'))
+    params['CustomerGatewayIds'] = module.params.get('customer_gateway_ids')
+
+    try:
+        result = json.loads(json.dumps(connection.describe_customer_gateways(**params), default=date_handler))
+    except Exception as e:
+        module.fail_json(msg=str(e.message))
+    snaked_customer_gateways = [camel_dict_to_snake_dict(gateway) for gateway in result['CustomerGateways']]
+    if snaked_customer_gateways:
+        for customer_gateway in snaked_customer_gateways:
+            if 'tags' in customer_gateway.keys() and customer_gateway['tags'] != []:
+                for tag_item in customer_gateway['tags']:
+                    if tag_item['key'] == 'Name':
+                        customer_gateway['customer_gateway_name'] = tag_item['value']
+    module.exit_json(changed=False, CustomerGateways=snaked_customer_gateways)
+
+
+def main():
+
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(
+        dict(
+            customer_gateway_ids=dict(default=[], type='list'),
+            filters=dict(default={}, type='dict')
+        )
+    )
+
+    module = AnsibleModule(argument_spec=argument_spec,
+                           mutually_exclusive=[
+                               ['customer_gateway_ids', 'filters']
+                           ],
+                           supports_check_mode=True
+                           )
+
+    if not HAS_BOTO3:
+        module.fail_json(msg='boto3 required for this module')
+
+    region, ec2_url, aws_connect_params = get_aws_connection_info(module, boto3=True)
+
+    if region:
+        connection = boto3_conn(module, conn_type='client', resource='ec2', region=region, endpoint=ec2_url, **aws_connect_params)
+    else:
```

Comment:
No need to check `region` before using `boto3_conn` - `boto3_conn` will handle a `NoRegionError` for you.